### PR TITLE
Add TypeScript support

### LINKED ISSUE #305: TypeScript support
I would love to have Typescript syntax highlighting in gitlab. Typescript is a superset of Javascript. I could try to add Typescript to rouge, but that is probably a 90% copy-paste activity. What would be the best way to proceed?

### PR DESCRIPTION
**Pull request by Seikho**
Aims to resolve #305 

I've done this pretty naively (I don't use Ruby at all) to add (at least) basic support for TypeScript.

Thanks

### CODE REVIEW

**Comment by PascalPrecht** [comment]
@jneen What's the status of this?

Is this planned to land anytime soon?

**Comment by sclausen** [comment]
+1

**Comment by duyaokun** [comment]
:+1:

**Comment by sanewell92** [comment]
image

**Comment by leocaseiro** [comment]
@jneen please accept this PR

**Comment by Turbots** [comment]
@jneen Agreed, please merge the PR 👍

**Comment by lmmfranco** [comment]
:+1:

**Comment by devilsray** [comment]
+1

**Comment by lpeterse** [comment]
+1

**Comment by leocaseiro** [comment]
Let's ping @jneen. She said [she was happy to merge](https://github.com/jneen/rouge/issues/305#issuecomment-150253956) that....

It's a shame [gitlab](https://gitlab.com/gitlab-org/gitlab-ce/issues/2914) depends on this lib. Almost 4 months and nothing....

**Comment by fenduru** [comment]
:+1:

**Comment by leocaseiro** [comment]
please @jneen, what is this status???
need correction or just press the button accept?

**Comment by leocaseiro** [comment]
I hate being annoying, but no feedback is more annoying than +1 or ping...

So, ping @jneen

**Comment by leocaseiro** [comment]
Feb
March
Apr
May
June
....

**Comment by Seikho** [comment]
There is a smidge of additional info [here](https://gitlab.com/gitlab-org/gitlab-ce/issues/13372)

**Comment by jneen** [comment]
Hi @leocaseiro, I'm getting to it, I appreciate your patience. I'll also refer you to #411 for an update about what's been going on these last months and the future of the project.

File: spec/visual/samples/typescript [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    constructor() {
+        super();
+    }
+    
+    readonly foo: number = 42;
+}
+
+async function fooAsync() {
+    let result = await barAsync();
+}
```
Comment (jneen) [inline spec/visual/samples/typescript]: Please add a newline at the end of the file.

**Comment by jneen** [comment]
Other than that, this is a very well-put-together lexer, awesome work!

**Comment by leocaseiro** [comment]
Hi @jneen, I wasn't aware of that. I'm really sorry.

I have some libraries open source, and I'm quite well-known how hard is to keep projects up to date. Specially these we code as volunteers. That's why I added a couple of contributors with merge/code permission. My bigger project has less than a third of starts your project has and contributors help me a lot. I've jus read [gitlab hired you](https://gitlab.com/gitlab-org/gitlab-ce/issues/13372#note_12117127)

Unfortunately I've never done any Ruby, otherwise, I'd volunteer myself to help you here.

I'm very happy for your

**Comment by Seikho** [comment]
Thanks, @jneen .

**Comment by pleerock** [comment]
awesome! when this will be released?

**Comment by jneen** [comment]
This week sometime, hopefully.

**Comment by jneen** [comment]
Released in v1.11.0.

**Comment by leocaseiro** [comment]
Yeaaaaaaaa 🍺 🍻 🎉

**Comment by PascalPrecht** [comment]
Is this now also deployed on gh-pages as well? Does anyone know?

**Comment by pleerock** [comment]
if [this](https://pages.github.com/versions/) is always up to date, then it seems that no. Maybe need to create a request on github?

**Comment by stanhu** [comment]
GitLab just deployed v8.9RC1, which has Rouge v1.11.0. As you can see here, TypeScript is now supported: https://gitlab.com/gitlab-org/gitlab-ce/snippets/20670

**Comment by Turbots** [comment]
In some cases, I still get colons being marked in red (as an error)... First thing that I notice is that it's usually in lines with multiple colons in it.

**Comment by Seikho** [comment]
Some es6 features require support in the JavaScript lexer. I didn't implement support for generics and casts.

**Comment by leocaseiro** [comment]
Thank you so much @Seikho, @jneen and any others who make this happened. Yesterday the gitlab 8.9 was released and now we have syntax highlighter for typescript.

I'm very excited about that.

🍻  🎉

**Pull request merged**